src/main/java/com/thealgorithms/dynamicprogramming/Knapsack.java

```diff
@@ -1,38 +1,51 @@
 package com.thealgorithms.dynamicprogramming;
 
 /**
- * A DynamicProgramming based solution for 0-1 Knapsack problem
+ * A Dynamic Programming based solution for the 0-1 Knapsack problem.
+ * This class provides a method, `knapSack`, that calculates the maximum value that can be
+ * obtained from a given set of items with weights and values, while not exceeding a
+ * given weight capacity.
+ *
+ * @see <a href="https://en.wikipedia.org/?title=0-1_Knapsack_problem">0-1 Knapsack Problem </a>
  */
-public class Knapsack {
+public final class Knapsack {
 
-    private static int knapSack(int W, int[] wt, int[] val, int n) throws IllegalArgumentException {
-        if (wt == null || val == null) {
-            throw new IllegalArgumentException();
+    private Knapsack(){}
+    ;
+
+    /**
+     * Solves the 0-1 Knapsack problem using Dynamic Programming.
+     *
+     * @param weightCapacity The maximum weight capacity of the knapsack.
+     * @param weights        An array of item weights.
+     * @param values         An array of item values.
+     * @return The maximum value that can be obtained without exceeding the weight capacity.
+     * @throws IllegalArgumentException If the input arrays are null or have different lengths.
+     */
+    public static int knapSack(final int weightCapacity, final int[] weights, final int[] values) throws IllegalArgumentException {
+        if (weightCapacity < 0) {
+            throw new IllegalArgumentException("Weight capacity should not be negative.");
+        }
+        if (weights == null || values == null || weights.length != values.length) {
+            throw new IllegalArgumentException("Input arrays must not be null and must have the same length.");
+        }
+        for (int w : weights) {
+            if (w <= 0) {
+                throw new IllegalArgumentException("Input array should not contain non-positive weight(s).");
+            }
         }
-        int i, w;
-        int[][] rv = new int[n + 1][W + 1]; // rv means return value
 
-        // Build table rv[][] in bottom up manner
-        for (i = 0; i <= n; i++) {
-            for (w = 0; w <= W; w++) {
-                if (i == 0 || w == 0) {
-                    rv[i][w] = 0;
-                } else if (wt[i - 1] <= w) {
-                    rv[i][w] = Math.max(val[i - 1] + rv[i - 1][w - wt[i - 1]], rv[i - 1][w]);
-                } else {
-                    rv[i][w] = rv[i - 1][w];
+        // DP table to store the state of the maximum possible return for a given weight capacity.
+        int[] dp = new int[weightCapacity + 1];
```

Comment:
If you insist on having this comment, why not to create a method like `private static int[] initializeDpArray(final int weightCapacity)`;

If you decide to have it, please check here if `weightCapacity >= 0` (not in the method `checkInput` mentioned earlier).

Reply:
Pardon @vil02  , which comment you are talking about?

Reply:
Sorry @vil02, looks like already replied to your comment but it was in Pending state. Please check my [here](https://github.com/TheAlgorithms/Java/pull/4368#discussion_r1328132121)
Please tell if any changes needed

Reply:
> I belive creating other method initializeDpArray will be overkill,

Understood.

> and why can't we check weightCapacity < 0 in our validation method?

If the method `initializeDpArray` would be introduced, I would suggest to make the `weightCapacity < 0` there.